dev/NavigationView/NavigationViewItemCollapsedEventArgs.cpp

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NavigationViewItemCollapsedEventArgs.h"
+
+winrt::NavigationViewItemBase NavigationViewItemCollapsedEventArgs::CollapsedItemContainer()
+{
+    return m_collapsedItemContainer.get();
+}
+
+void NavigationViewItemCollapsedEventArgs::CollapsedItemContainer(winrt::NavigationViewItemBase const& value)
+{
+    m_collapsedItemContainer.set(value);
+}
+
+winrt::IInspectable NavigationViewItemCollapsedEventArgs::CollapsedItem()
+{
+    if (auto const nv = m_navigationView.get())
+    {
+        return nv.MenuItemFromContainer(m_collapsedItemContainer.get());
+    }
+    return nullptr;
+}
+
+void NavigationViewItemCollapsedEventArgs::NavigationView(winrt::NavigationView const& navigationView)
```

Comment:
>NavigationView [](start = 43, length = 14)

I think it would be better to delete this and add a navigationView parameter to the constructor so you don't forget to do this.

Reply:
Maybe that isn't safe?

---
In reply to: [401134140](https://github.com/microsoft/microsoft-ui-xaml/pull/2196#discussion_r401134140) [](ancestors = 401134140)

Reply:
Moved to constructor. What makes you say this would not be safe?

---
In reply to: [401134836](https://github.com/microsoft/microsoft-ui-xaml/pull/2196#discussion_r401134836) [](ancestors = 401134836,401134140)

Reply:
Sorry, put that on the wrong comment, maybe it isn't safe to cache the result of MenuItemFromContainer call. I still don't have my head around how recycling works here.

Reply:
This should be fine either way. We create a new args for every event and the lookup to get from container to item is quite efficient so caching is not really necessary in this case.